lib/index.js

```diff
@@ -0,0 +1,187 @@
+const validate = require('bedrock-validation').validate;
+const BedrockError = bedrock.util.BedrockError;
+
+let strategy = null;
+
+// module API
+const api = {};
+module.exports = api;
+
+const logger = bedrock.loggers.get('app');
+
+require('./config');
+
+// configure passport before serving static files
+bedrock.events.on('bedrock-express.configure.static', (e, callback) => {
+  const DidJwtStrategy = require('./DidJwtStrategy');
+  try {
+    strategy = new DidJwtStrategy();
+  } catch(err) {
+    logger.error('Error when attempting to instantiate DidJwtStrategy.', err);
+    return callback(err);
+  }
+  bedrock.events.emit(
+    'bedrock-authn-did-jwt.config.keyStore', strategy, err => {
+      if(err) {
+        logger.error('Error when attempting to provision JWT key store.', err);
+        return callback(err);
+      }
+      brPassport.use({
+        strategy: strategy
+      });
+      callback();
+    });
+});
+
+bedrock.events.on('bedrock.ready', callback => strategy.init(callback));
+
+// add routes
+bedrock.events.on('bedrock-express.configure.routes', app => {
+  const routes = config['authn-did-jwt'].routes;
+
+  app.post(routes.login, api.login);
+
+  // only add route for generating tokens if there is a key store
+  if(strategy.store) {
+    app.post(routes.token, optionallyAuthenticated, (req, res, next) => {
+      if(req.user && 'identity' in req.user) {
+        const payload = {
+          sub: req.user.identity.id
+        };
+        if(req.body.aud) {
+          if(req.body.aud.indexOf('https:') !== 0) {
+            return next(new BedrockError(
+              '`aud` must be an HTTPS URL.', 'InvalidAud',
+              {aud: req.body.aud, public: true, httpStatusCode: 400}));
+          }
+          payload.aud = req.body.aud;
+        }
+        return api.generateToken(payload, (err, token) => {
+          if(err) {
+            return next(err);
+          }
+          res.json({token: token});
+        });
+      }
+      const v = validate('authn-did-jwt.postToken', req.body);
+      if(!v.valid) {
+        return next(v.error);
+      }
+      const email = req.body.email.toLowerCase();
+      async.auto({
+        getIdentity: callback => {
+          brIdentity.getAll(
+            null, {'identity.email': email}, {
+              'identity.id': true,
+              'identity.email': true
+            }, (err, result) => {
+              if(err) {
+                return callback(err);
+              }
+              if(result.length === 0) {
+                return callback(new BedrockError(
+                  'Identity not found.', 'NotFound',
+                  {email: email, public: true, httpStatusCode: 404}));
+              }
+              callback(null, result[0].identity);
+            });
+        },
+        generateToken: ['getIdentity', (callback, results) =>{
+          const did = results.getIdentity.id;
+          api.generateToken({sub: did}, callback);
```

Comment:
Note: This probably doesn't need to be limited to a DID... could be an HTTPS ID as well... but let's not worry about that right this minute.